wave_benchmark/src/trajectory_compare.cpp

```diff
@@ -0,0 +1,44 @@
+#include "wave/benchmark/trajectory_compare.hpp"
+
+namespace wave {
+
+void TrajectoryCompare::reset() {
+    this->ground_truth.clear();
+    this->measurements.clear();
+    this->error.clear();
+}
+
+void TrajectoryCompare::push_measurement(Pose pose, const TimeType time) {
+    measurements.insert(PoseMeasurement(time, this->measurement_key, pose));
+}
+
+void TrajectoryCompare::push_truth(Pose pose, const TimeType time) {
+    ground_truth.insert(PoseMeasurement(time, this->ground_truth_key, pose));
+}
+
+void TrajectoryCompare::calculate_error() {
```

Comment:
Yes it could be a measurement container. It could also be something more flexible like a pair of iterators for each sequence.

Sure, 1-by-1 iteration makes sense but it is simpler for the user (and much simpler for you, the maintainer) to build up a standard container first. It is also much simpler in the plausible event that the user already has a container.